tests/test_zk_evm.py

```diff
@@ -0,0 +1,835 @@
+    },
+    {
+        "params": {
+            "code": "6001600757600a5b600a6000600857600a01",
+            "calldata": "",
+            "stack": "20",
+            "memory": "",
+            "return_value": "",
+        },
+        "id": "JUMP if condition is met",
+        "marks": pytest.mark.skip("Returned stack is 10,20 instead of 20"),
+    },
+    {
+        "params": {
+            "code": "601160405200",
+            "calldata": "",
+            "stack": "",
+            "memory": "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000011",
+            "return_value": "",
+        },
+        "id": "Memory operations - Check very large offsets",
+    },
+    {
+        "params": {
+            "code": "6011604052602260405200",
+            "calldata": "",
+            "stack": "",
+            "memory": "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000022",
+            "return_value": "",
+        },
+        "id": "Memory operations - Check Colliding offsets",
+    },
+    {
+        "params": {
+            "code": "7d111111111111111111111111111111111111111111111111111111111111600052",
+            "calldata": "",
+            "stack": "",
+            "memory": "0000111111111111111111111111111111111111111111111111111111111111",
+            "return_value": "",
+        },
+        "id": "Memory operations - Check saving memory with 30 bytes",
+    },
+    {
+        "params": {
+            "code": "7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff604052601160355200",
+            "calldata": "",
+            "stack": "",
+            "memory": "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000011ffffffffffffffffffffff",
+            "return_value": "",
+        },
+        "id": "Memory operations - Check saving memory in between an already saved memory location",
+    },
+]
+
+
+params = [pytest.param(*Params(**case.pop("params")), **case) for case in test_cases]
+
+
+@pytest.mark.asyncio
+class TestZkEVM:
+    @staticmethod
+    def int_to_uint256(value):
+        low = value & ((1 << 128) - 1)
+        high = value >> 128
+        return low, high
+
+    @pytest.mark.parametrize(
+        argnames,
+        params,
+    )
+    async def test_case(self, zk_evm, code, calldata, stack, memory, return_value):
+        Uint256 = zk_evm.struct_manager.get_contract_struct("Uint256")
+        res = await zk_evm.execute(
+            code=[int(b, 16) for b in wrap(code, 2)],
+            calldata=[int(b, 16) for b in wrap(calldata, 2)],
+        ).execute(caller_address=1)
```

Comment:
```suggestion
        ).call(caller_address=1)
```
Switching to call will probably improve the test speed as it won't imply any state change nor hash computing

Reply:
trying this rn